src/docs/asciidoc/Modules.asciidoc

```diff
@@ -264,6 +265,22 @@ In cases where you want to start e.g. a single-node runtime with a custom module
 `XD_CUSTOMMODULE_HOME=file\:/path/to/custom-modules bin/xd-singlenode`
 ====
 
+[[replicating-module-registry]]
+==== Replicating Module Registry
+When running in distributed mode, an alternative to using a shared file system for custom modules is to use the _replicating module registry_.
+
+If the value of `xd.customModule.home` does not use the `file:` protocol, then Spring XD will automatically set up a replicating registry that proxies that remote registry (at the end of the day, all modules need to be read from local filesystem anyway). This is all done transparently and by default, files are copied down from the central repository only if their contents has changed.
+
+At the time of writing, only the `hdfs:` protocol is supported. Setting this up is straightforward:
+
+[source,yml]
+----
+xd:
+  customModule:
+    home: hdfs://somehost/root/path/of/registry
+----
+
+Files will be replicated on the local filesystem in a temporary directory, on demand and loaded from there. Obviously, the XD Admin process will need to have write access to that shared HDFS directory. Intermediary paths (`/root/path/of/registry` in the example above) are created at startup if they don't exist yet.
```

Comment:
You're right, even more so because at this time, the connection details to HDFS are very basic, so it's not even "correct" (as in there is no authentication against the unix user running the process)